OpenTrons/opentrons-api
# chore(robot-server, api): filter fixit commands out of the commands list

# Overview

reference https://github.com/Opentrons/opentrons/pull/16001 for previous pr comments.
closes [EXEC-654](https://opentrons.atlassian.net/browse/EXEC-654).
get filtered command list for historical runs and for current runs.

## Test Plan and Hands on Testing

- upload a protocol that will enter ER.
- add some fixit commands.
- make sure you dont see the fixit commands in the run log. 
- when the run is completed and un-current make sure you still dont see fixit commands. 

## Changelog

- added an arg `includeFixitCommands` to `get_run_commands` route.
- add logic to `CommandState` and `CommandHistory` to filter based on the protocol intent. 
- added column `command_intent` to `run_command_table` and migration scripts.
- fixed but with copying over enum columns in the DB. 
- added filtering in the sql query for command intent. 

## Review requests

changes make sense? how ugly is this? 

## Risk assessment

Medium. need to make sure run commands did not changed the default behavior. 

[EXEC-654]: https://opentrons.atlassian.net/browse/EXEC-654?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
